api/src/main/java/org/openmrs/module/patientqueueing/api/impl/PatientQueueingServiceImpl.java

```diff
@@ -78,4 +145,22 @@ public PatientQueue getIncompletePatientQueue(Patient patient, Location location
 		return dao.getIncompletePatientQueue(patient, locationTo);
 	}
 	
+	/**
+	 * This Method is a helper method to the generateQueueNumber which checks if the queueNumber
+	 * generated exists
+	 * 
+	 * @param queueNumber A queueNumber in form of a string which will be check to determine if it
+	 *            exists
+	 * @return
+	 */
+	private boolean isQueueNumberIdExisting(String queueNumber) {
+		PatientQueue patientQueue = getPatientQueueByQueueNumber(queueNumber);
```

Comment:
Wouldn't be simpler to just `return getPatientQueueByQueueNumber(queueNumber) != null`?